File: app/src/main/java/com/allever/app/ad/AdFactory.java
package com.allever.app.ad;

import android.content.Context;

//import com.allever.lib.ad.admob.MobAdmobBan;
//import com.allever.lib.ad.admob.MobAdmobInter;
import com.allever.lib.ad.facebook.MobFacebookBan;
import com.allever.lib.ad.facebook.MobFacebookInter;
import com.mob.core.MobBannerAd;
import com.mob.core.MobInterstitialAd;
import com.mob.main.IMobAdFactory;

/**
 * @author allever
 */
public class AdFactory implements IMobAdFactory {
    @Override
    public MobBannerAd getBannerAd(Context context, String cate, String pub, String appid) {
        MobBannerAd ad  =null;
        switch (cate) {
            case "F":
                ad = new MobFacebookBan(context,pub);
                break;
            case "A":
//                ad = new MobAdmobBan(context,pub);
                break;
        }
        return ad;
    }

    @Override
    public MobInterstitialAd getInterstitialAd(Context context, String cate, String pub, String appid) {
        MobInterstitialAd ad  =null;
        switch (cate) {
            case "F":
                ad = new MobFacebookInter(context,pub);
                break;
            case "A":
//                ad = new MobAdmobInter(context,pub);
                break;
        }
        return ad;
    }
}

File: app/src/main/java/com/allever/app/ad/AdFactory.kt
package com.allever.app.ad

import com.allever.lib.ad.AdBusiness
import com.allever.lib.ad.admob.AdMobBusiness

import com.allever.lib.ad.chain.IAdBusiness
import com.allever.lib.ad.chain.IAdBusinessFactory


/**
 * @author allever
 */
class AdFactory : IAdBusinessFactory {

    override fun getAdBusiness(businessName: String): IAdBusiness? {
        return when(businessName) {
            AdBusiness.A -> {
                AdMobBusiness
            }
            else -> {
                null
            }
        }
    }
}
